packages/next-international/src/app/client/create-use-change-locale.ts

```diff
@@ -24,8 +30,12 @@ export function createUseChangeLocale<LocalesKeys>(useCurrentLocale: () => Local
     }
 
     return function changeLocale(newLocale: LocalesKeys) {
-      push(`/${newLocale}${pathWithoutLocale}${finalSearchParams}`);
-      refresh();
+      locales[newLocale as keyof typeof locales]().then(module => {
+        localesCache.set(newLocale as string, module.default);
+
+        push(`/${newLocale}${pathWithoutLocale}${finalSearchParams}`);
+        refresh();
+      });
     };
```

Comment:
Maybe we could use our Cache here instead of process the promise everytime? Overusing promises bothers me a bit. I don't find it very natural.

Reply:
After the first `import()`, the next one won't re-fetch again the locale file because it's in the cache. The overhead of running the promise is **very** minimal.

Reply:
Oh, okay. I guess it's because the promise is already consumed?

Reply:
I've just thought again about my suggestion. To be honest, it made sense in my approach where I called `flattenLocales` when caching. But given the current implementation choices, it doesn't make much sense.

Please, ping me when the PR consisting of refactoring calls to the `flattenLocales` function will be drafted. I haven't looked at the lib code much, but from what little I've read of it to help achieve better behaviour when switching languages on the client side, I've really liked it, both in terms of the typing and the way the code is structured and concise.

I'm curious.